tests/dialects/test_tsql.py

```diff
@@ -446,6 +501,7 @@ def test_udf(self):
             pretty=True,
         )
 
+    def test_fullproc(self):
```

Comment:
Why was this moved into a different testing method?

Reply:
I think I'd just move `BEGIN`, `END`, `SET XACT_ABORT ON` to `test_tsql`. I'm not sure if the `DECLARE` one is udf-related, but if it isn't I'd move that there too. We can keep `test_fullproc` as is, maybe rename to a non-abbreviated form like `test_procedure_definition`?